controller/src/main/java/io/pravega/controller/server/rpc/auth/RESTAuthHelper.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.rpc.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import io.pravega.auth.AuthException;
+import io.pravega.auth.AuthHandler;
+import io.pravega.auth.AuthenticationException;
+import io.pravega.auth.AuthorizationException;
+
+import javax.ws.rs.core.Response;
+import java.security.Principal;
+import java.util.List;
+
+/**
+ * Helper class for handling auth (authentication and authorization) for the REST API.
+ */
+public class RESTAuthHelper {
+
+    private final PravegaAuthManager pravegaAuthManager;
+
+    public RESTAuthHelper(PravegaAuthManager pravegaAuthManager) {
+        this.pravegaAuthManager = pravegaAuthManager;
+    }
+
+    public boolean isAuthorized(List<String> authHeader, String resourceName, Principal principal,
+                                AuthHandler.Permissions permissionLevel)
+            throws AuthException {
+        if (isAuthEnabled()) {
+            return pravegaAuthManager.authorize(resourceName,
+                    principal,
+                    parseCredentials(authHeader),
+                    permissionLevel);
+        } else {
+            // Since auth is disabled, every request is deemed to have been authorized.
+            return true;
+        }
+    }
+
+    public void authorize(List<String> authHeader, String resourceName, Principal principal,
+                          AuthHandler.Permissions permissionLevel) throws AuthException {
+        if (!isAuthorized(authHeader, resourceName, principal, permissionLevel)) {
+            throw new AuthorizationException(
+                    String.format("Failed to authorize for resource [%s]", resourceName),
+                    Response.Status.FORBIDDEN.getStatusCode());
+        }
+    }
+
+    public  Principal authenticate(List<String> authHeader) throws AuthException {
```

Comment:
unnecessary extra space after `public`

Reply:
Hmm.. you certainly have a keen eye! :) Will fix it.